### PR TITLE
Add tag validation to application

### LINKED ISSUE #13: Required Tag Association
#### As a user I want all images to have at least one tag.

__Story__:

The tag feature has been a hit. Our users love being able to filter images by
cats, SR-71s, and "AppFolio tshirts" to name a few. In fact, we have discovered
that images without any tags receive significantly less views than those with
tags. As a result, we have decided to require all images to have at least one
tag.

__Acceptance criteria__:
- [x] I cannot save an image without adding at least one tag.
- [x] An error message is associated with the appropriate input field when a
tag is not provided.
- [x] All images have at least one tag.

__Discussion Topic__:
- [ ] What do you do about existing images that have no tags?

__Dependencies__:
- Image Tags
- Refactor flow tests to use page objects

### PR DESCRIPTION
**Pull request by mathewbaltes**
Closes #13

### CODE REVIEW

File: test/models/images_test.rb [hunk 1]
```diff
@@ -20,15 +20,15 @@ class ImageModelTest < ActiveSupport::TestCase
     assert_not_empty image.errors
   end
 
-  test 'verify image with null tag is valid' do
+  test 'verify image with null tag is invalid' do
     image = Image.new(url: 'http://www.appfolio.com/', tag_list: nil)
-    assert_equal true, image.valid?
-    assert_empty image.errors
+    assert_not image.valid?, 'Image with no tags is invalid'
```
Comment (pololee) [inline test/models/images_test.rb]: You may also want to test your customized error message `One or more tags is required`

File: test/models/images_test.rb [hunk 2]
```diff
@@ -20,15 +20,15 @@ class ImageModelTest < ActiveSupport::TestCase
...
+    assert_not_empty image.errors
   end
 
-  test 'verify image with empty tag is valid' do
+  test 'verify image with empty tag is invalid' do
     image = Image.new(url: 'http://www.appfolio.com/', tag_list: '')
-    assert_equal true, image.valid?
-    assert_empty image.errors
+    assert_not image.valid?, 'Image with no tags is invalid'
+    assert_not_empty image.errors
```
Comment (pololee) [inline test/models/images_test.rb]: Same as the above comment.
> You may also want to test your customized error message `One or more tags is required`

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -106,7 +94,7 @@ def setup
   end
 
   test 'deleting a shared image shows success if successful' do
-    Image.create(url: 'http://test.com/')
+    Image.create(url: 'http://test.com/', tag_list: '123')
```
Comment (pololee) [inline test/controllers/images_controller_test.rb]: This test file can use some refactor.
You can have a private method like this

```ruby
private

def image_params(url: 'http://test.com/', tag_list: '123')
  {
    url: url,
    tag_list: tag_list
  }
end
```

Comment (mathewbaltes) [inline test/controllers/images_controller_test.rb]: Created a factory instead, thanks!

**Pull request merged**